Automattic/newspack-plugin
# Use WPCOM as a proxy for Google OAuth2 flow

### All Submissions:






### Changes proposed in this Pull Request:



Changes how Google OAuth flow works to a proxy flow.

### How to test the changes in this Pull Request:

1. Ensure you're authenticated with WPCOM by visiting the Support Wizard
2. Add a `NEWSPACK_GOOGLE_OAUTH_ENABLED` env variable of value `true` - this is effectively a feature flag that will enable us to test on production
4. To start afresh, remove `_newspack_google_oauth` user meta and revoke "Newspack" app access at https://myaccount.google.com/permissions
5. Visit the Newspack dashboard, click on the "Google OAuth2" tile and complete the auth flow. At the end the tile should feature your email address
6. Revoke access, complete the flow again
7. Remove the credentials from the DB (the user meta), complete the flow again
8. Wait for at least an hour (so the credentials expire), load the dashboard and observe the token was refreshed - the site can still fetch the email address

### Other information:
